pkg/saga/statemachine/engine/store/db/db.go

```diff
@@ -0,0 +1,102 @@
+package db
+
+import (
+	"database/sql"
+	"github.com/pkg/errors"
+	"github.com/seata/seata-go/pkg/util/log"
+)
+
+const TimeLayout = "2006-01-02 15:04:05.999999999-07:00"
+
+type ExecStatement[T any] func(obj T, stmt *sql.Stmt) (int64, error)
+
+type ScanRows[T any] func(rows *sql.Rows) (T, error)
+
+type Store struct {
+	db *sql.DB
+}
+
+func SelectOne[T any](db *sql.DB, sql string, fn ScanRows[T], args ...any) (T, error) {
```

Comment:
Generics are also used here, need to remove them?

Reply:
I suggest we can keep this generic function. It maps the `Rows` to arbitrary type `T`, which I think it is a good abstract in my POV. If not using generics, we can only change the return type to `interface{}`, and use type conversion `db.SelectOne(...).(StateMachineInstance)` at the invocation.